src/Cryptol/Parser/NoPat.hs

```diff
@@ -98,11 +99,11 @@ noPat pat =
          return (pTy r x ty, zipWith getN as [0..] ++ concat dss)
 
     PRecord fs ->
```

Comment:
This part of the code is maybe one I understand the least.  This change causes the fields to be processed in canonical order instead of display order.  I don't think that's a problem, but I'm not as sure as with most places.

Reply:
I'm pretty sure that's not a problem. The order-sensitive effects in the `NoPatM` monad are the unique name supply and the ordering of error messages. But really no other code should care about permutations of the unique name ids, or of the ordering of `NoPat` errors, so for all intents and purposes I think we can treat `NoPatM` like a commutative monad.